cli/src/commands/init.rs

```diff
@@ -90,9 +117,79 @@ fn prompt_group() -> io::Result<Option<String>> {
         return Ok(None);
     }
 
-    let group: String = Input::with_theme(&ColorfulTheme::default())
-        .with_prompt("Project Group (default: none)")
-        .interact_text()?;
+    let group: String =
+        Input::new().with_prompt("Project Group (default: none)").interact_text()?;
 
     Ok(Some(group))
 }
+
+/// Interactively ask for missing lockfile information.
+fn prompt_lockfile(
+    cli_lockfile: Option<&String>,
+    cli_lockfile_type: Option<&String>,
+) -> io::Result<(String, Option<String>)> {
+    if let Some(lockfile) = cli_lockfile {
+        return Ok((lockfile.clone(), cli_lockfile_type.cloned()));
+    }
+
+    // Prompt for lockfile name.
+    let lockfile = prompt_lockfile_name()?;
+
+    // Try to determine lockfile name from known lockfiles.
+    for format in LockfileFormat::iter() {
+        if format.parser().is_path_lockfile(Path::new(&lockfile)) {
+            let lockfile_type = format.name().to_owned();
+            return Ok((lockfile, Some(lockfile_type)));
+        }
+    }
+
+    // Prompt for lockfile type.
+    let lockfile_type = prompt_lockfile_type()?;
+
+    Ok((lockfile, Some(lockfile_type)))
+}
+
+/// Ask for the lockfile name.
+fn prompt_lockfile_name() -> io::Result<String> {
+    // Find all known lockfiles in the currenty directory.
+    let mut lockfiles: Vec<_> = fs::read_dir("./")?
+        .into_iter()
+        .flatten()
+        .filter(|entry| {
+            LockfileFormat::iter().any(|format| format.parser().is_path_lockfile(&entry.path()))
+        })
+        .flat_map(|entry| entry.file_name().to_str().map(str::to_owned))
+        .collect();
+
+    // Prompt if a lockfile was found.
+    if !lockfiles.is_empty() {
+        // Add choice to specify an unknown lockfile.
+        lockfiles.push(String::from("other"));
+
+        // Ask user for lockfile.
+        let index = FuzzySelect::new()
+            .with_prompt("Select your project's lockfile")
+            .items(&lockfiles)
+            .interact()?;
+
+        // Return selected lockfile unless `other` was chosen.
+        if index + 1 != lockfiles.len() {
+            return Ok(lockfiles.remove(index));
+        }
+    }
+
+    // Prompt for lockfile name if none was selected.
+    Input::new().with_prompt("Project lockfile name").interact_text()
+}
+
+/// Ask for the lockfile type.
+fn prompt_lockfile_type() -> io::Result<String> {
+    let lockfile_types: Vec<_> = LockfileFormat::iter().map(|format| format.name()).collect();
+
+    let index = FuzzySelect::new()
+        .with_prompt("Select lockfile's type")
+        .items(&lockfile_types)
+        .interact()?;
+
+    Ok(lockfile_types[index].to_owned())
+}
```

Comment:
With command line arguments, it's possible to set the lockfile type to `None`. But that is not an option in the prompt. This list doesn't even include `auto` as an option.

I think we should stay consistent. If we are okay with `None` (or `auto`) as an option, we should add it to this list. If we are not okay with it, then it should be an error when command line arguments would result in a lockfile type of `None`.

Reply:
I think the difference makes a lot of sense. You might not care/want to specify it when running an analyze because you trust Phylum to detect it automatically. But when configuring your project there is one guaranteed lockfile type that is going to be correct and that shouldn't be left to be figured out at a later point.

We're already trying to detect it automatically based on file name, but if that isn't enough I think we should definitely force the user to specify which format it actually is.

And I also think that our existing analyze/parse CLI should stay with an auto option.

Reply:
> But when configuring your project there is one guaranteed lockfile type that is going to be correct and that shouldn't be left to be figured out at a later point.

I agree with that. But by that logic, `phylum init` should never set the lockfile type to `None`. So passing `--lockfile` without `--lockfile-type` should error if we can't detect the lockfile type.